src/sage/rings/semirings/tropical_variety.py

```diff
@@ -0,0 +1,1124 @@
+                        vertex = [R(e.subs(v==right)) for e in line[0]]
+                        vertices[index].add(tuple(vertex))
+
+            # find the interval of parameter for outer vertex
+            interval1 = RealSet(-infinity,infinity)  # represent t1
+            interval2 = RealSet(-infinity,infinity)  # represent t2
+            is_doublevar = False
+            for i, point in enumerate(comps[index][0]):
+                vs = point.variables()
+                if len(vs) == 1:
+                    temp1 = RealSet(solve(point>=axes[i][0], vs[0])[0][0])
+                    temp2 = RealSet(solve(point<=axes[i][1], vs[0])[0][0])
+                    temp = temp1.intersection(temp2)
+                    if vs[0] == vars[0]:
+                        interval1 = interval1.intersection(temp)
+                    else:
+                        interval2 = interval2.intersection(temp)
+                elif len(vs) == 2:
+                    sol1 = solve(point>=axes[i][0], vs)
+                    sol2 = solve(point<=axes[i][1], vs)
+                    is_doublevar = True
+
+            # calculate the outer vertex with t1 fixed
+            for p in [interval1.inf(), interval1.sup()]:
+                new_param = [e.subs(vars[0]==p) for e in comps[index][1]]
+                temp = solve(new_param, vars[1])
+                if temp:
+                    interval_param = RealSet()
+                    for t in temp:
+                        if t != []:
+                            interval_param = interval_param + RealSet(t[0])
+                        else:
+                            interval_param = interval_param + RealSet(-infinity, infinity)
+                    interval_param = interval_param.intersection(interval2)
+                    if is_doublevar:
+                        int1 = RealSet()
+                        for s1 in sol1:
+                            subs1 = solve(s1[0].subs(vars[0]==p), vars[1])
+                            try:
+                                int1 = int1 + RealSet(subs1[0])
+                            except TypeError:
+                                int1 = int1 + RealSet(subs1[0][0])
+                        int2 = RealSet()
+                        for s2 in sol2:
+                            subs2 = solve(s2[0].subs(vars[0]==p), vars[1])
+                            try:
+                                int2 = int2 + RealSet(subs2[0])
+                            except TypeError:
+                                int2 = int2 + RealSet(subs2[0][0])
+                        final_int = int1.intersection(int2)
+                        interval_param = interval_param.intersection(final_int)
+                    if interval_param:
+                        vertex1 = [R(e.subs(vars[0]==p, vars[1]==interval_param.inf())) for e in comps[index][0]]
+                        vertex2 = [R(e.subs(vars[0]==p, vars[1]==interval_param.sup())) for e in comps[index][0]]
```

Comment:
```suggestion
                        vertex1 = [R(e.subs(**{vars[0]: p, vars[1]: interval_param.inf()})) for e in comps[index][0]]
                        vertex2 = [R(e.subs(**{vars[0]: p, vars[1]: interval_param.sup()})) for e in comps[index][0]]
```
And similar elsewhere.